src/main/java/lotto/domain/Rank.java

```diff
@@ -24,13 +24,13 @@ public enum Rank {
     public static Rank of(int matchCount, boolean isBonus) {
         return Arrays.stream(Rank.values())
                 .filter(rank -> rank.isSameMatchCount(matchCount))
-                .filter(rank -> rank.isSecond(isBonus))
+                .filter(rank -> !rank.isSecond() || isBonus)
```

Comment:
```suggestion
                .filter(rank -> rank.isSameMatchCount(matchCount))
                .filter(rank -> rank.isNotSecondRank(isBonus))
```

`isSameMatchCount`좋은 표현이네요.
복잡한 비교연산자의 의도를 드러내면 어떨까요?